### DISCUSSION TITLE
thunderbolt networking

### DISCUSSION THREAD

**Discussion by Sebastian-blu** [apohl79/audiogridder#860] [General]
Hey was wondering if I can just connect my i9 MacBook Pro to my M1 pro via a thunderbolt cable instead of a Ethernet connection for audiogridder. I saw there is thunderbolt networking for file share. It would be nice cause I have thunderbolt cables and no ethernet adapters. Thanks!

**Comment by JamesXXV** [comment]
Yeah that's alright, actually its much better than a 1 gig ethernet connection and thunderbolt is supposed to have like crazy low latency, in fact I wanted to do that in my setup too but thunderbolt pcie cards are expensive af for windows machines. Can you like share the results of your setup? like for example on the M1 Pro you may or may not see a single core crashing on a ethernet connection when you have 0 buffer in audiogridder and your audio buffer in daw is lower than 1024, I just wanna see if that's the same case with thunderbolt.

edit: in thunderbolt connections cable length matters. 0.8m thunderbolt 3 cable by apple seems to be passive if you need longer lengths you'll need an active cable which might go upto 2m in length. generally any 2m long thunderbolt cable will be active.

https://9to5mac.com/2018/08/23/back-to-the-mac-011-are-longer-thunderbolt-3-cables-slower-video/

**Comment by Sebastian-blu** [comment]
Nice that’s good to know its is faster, though I think for my setup I have to use a tb2 cable to get started. That’s what I have on hand. My main question becomes how do I make the computers see each other through thunderbolt connection. Will audio grinder just see the other computer?

I can report back once I have something going for sure. Honestly once all my acustica plugins are working on this m1pro I won’t need audiogridder and the old comp because this m1 pro can handle things brilliantly. 

Thanks!

**Reply by JamesXXV** [reply]
now how do I put this, tb4 is backwards compatible with tb3, tb3 is backwards compatible with tb2 but tb4 is NOT backward compatible with tb2 so even if you got a tb3 usb c to tb2 adapter your M1pro will not even detect the other 16 inch but there's no harm in trying tho. if that doesn't work you'll need to get a new tb3 or tb4 cable. Also you'll need to share your wifi to the second Mac(there is a setting inside System Preferences called internet sharing) using the tb connection for it to be able to use network for audiogridder.

edit:https://thunderbolttechnology.net/tech/faq

**Comment by manatactical** [comment]
I tried the TB3 network between a PC and a Mac. I got issues along the way to finally giving up. The DHCP was not working at all.
So I went for a 2.5G card on both and it's running pretty well (VEP7, Audio Gridder). Ping test shows sub .2 ms. Plenty of room for a lot of instances of AG and VEP instruments.

**Reply by JamesXXV** [reply]
and your Mac doesn't crash core audio when syncing live midi like say you're just playing a virtual synth loaded on audiogridder or anything of that sort? for me it has 2 block minimum buffer size otherwise it will crash core audio.

**Reply by manatactical** [reply]
I also use 2 blocks. The latency is awful with Logic (64 buffer), but quite low with Cubase (128 buffer but far better roundtrip latency) and overall very stable. It does work at 0 blocks but sometimes it causes stutter, even if I have monsters machines.
